dashboard/pages/Analytics.py

```diff
@@ -0,0 +1,416 @@
+"""Live analytics page for the StreamLit dashboard, showing live graph visualisations."""
+from datetime import datetime
+from os import environ
+from requests import get
+
+from dotenv import load_dotenv
+import pandas as pd
+
+import altair as alt
+from psycopg2 import extensions, connect
+import streamlit as st
+from vega_datasets import data
+
+# pylint: disable=E1136
+
+
+def get_db_connection() -> extensions.connection:
+    """Returns a connection to the AWS Bandcamp database"""
+    try:
+        return connect(user=environ["DB_USER"],
+                       password=environ["DB_PASSWORD"],
+                       host=environ["DB_IP"],
+                       port=environ["DB_PORT"],
+                       database=environ["DB_NAME"])
+    except ConnectionError:
+        print("Error: Cannot connect to the database")
+        return None
+
+
+def load_all_data(db_connection: extensions.connection) -> pd.DataFrame:
+    """Loads all the data from the database into a pandas dataframe"""
+    with db_connection.cursor() as curr:
+        curr.execute("""
+                    SELECT sale_event.*, country.country, item.item_name, item.item_type_id, item.item_image, artist.artist_name, genre.genre
+                    FROM sale_event
+                    JOIN country
+                    ON country.country_id = sale_event.country_id
+                    JOIN item
+                    ON item.item_id = sale_event.item_id
+                    JOIN artist
+                    ON artist.artist_id = item.artist_id
+                    JOIN item_genre
+                    ON item_genre.item_id = item.item_id
+                    JOIN genre
+                    ON genre.genre_id = item_genre.genre_id;""")
+        tuples = curr.fetchall()
+        column_names = ['sale_id', 'sale_time', 'amount', 'item_id',
+                        'country_id', 'country', 'item_name', 'item_type',
+                        'item_image', 'artist', 'genre']
+        return pd.DataFrame(tuples, columns=column_names)
+
+
+def build_date_range_slider() -> list[datetime]:
+    """Creates a slider for user to select data sample range; default range is the last 24 hours."""
+    return st.slider('Select Date Range:', min_value=pd.to_datetime(
+        '2024-01-08 00:00:00+00:00', utc=True).date(),
+        max_value=pd.to_datetime('now', utc=True).date(),
+        value=((pd.Timestamp('now', tz='UTC') - pd.Timedelta(days=1)
+                ).date(), pd.Timestamp('now', tz='UTC').date()),
+        format="DD/MM/YYYY",
+    )
+
+
+def create_sales_chart(df, object_type):
```

Comment:
Very minor but type hints would make this even better. Same goes to other functions